huaweicloud/services/waf/resource_huaweicloud_waf_rule_precise_protection.go

```diff
@@ -0,0 +1,554 @@
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the description of WAF precise protection rule.`,
+			},
+		},
+	}
+}
+
+func conditionsSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"field": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the field of the condition.`,
+			},
+			"logic": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the condition matching logic.`,
+			},
+			"subfield": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the subfield of the condition.`,
+			},
+			"content": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the content of the match condition.`,
+			},
+			"reference_table_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the reference table id.`,
+			},
+		},
+	}
+	return &sc
+}
+
+type updateRuleStatusOpt struct {
+	ProjectID string
+	PolicyID  string
+	RuleType  string
+	RuleID    string
+	Cfg       *config.Config
+}
+
+func resourceRulePreciseProtectionCreate(ctx context.Context, d *schema.ResourceData,
+	meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		preciseProtectionHttpUrl = "v1/{project_id}/waf/policy/{policy_id}/custom"
+		preciseProtectionProduct = "waf"
+	)
+	preciseProtectionClient, err := cfg.NewServiceClient(preciseProtectionProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating WAF Client: %s", err)
+	}
+
+	createPath := preciseProtectionClient.Endpoint + preciseProtectionHttpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", preciseProtectionClient.ProjectID)
+	createPath = strings.ReplaceAll(createPath, "{policy_id}", fmt.Sprintf("%v", d.Get("policy_id")))
+	createPath += buildQueryParams(d, cfg)
+	createOpt := golangsdk.RequestOpts{
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=utf8",
+		},
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+
+	bodyParam, err := buildCreateOrUpdateBodyParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createOpt.JSONBody = utils.RemoveNil(bodyParam)
+	createResp, err := preciseProtectionClient.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating RulePreciseProtection: %s", err)
+	}
+
+	createRespBody, err := utils.FlattenResponse(createResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("id", createRespBody)
+	if err != nil {
+		return diag.Errorf("error creating RulePreciseProtection: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	if v, ok := d.GetOk("status"); ok && v.(int) == 0 {
+		updateStatusOpt := updateRuleStatusOpt{
```

Comment:
the param of updateStatusOpt seems unnecessary,  ProjectID can be get from preciseProtectionClient, PolicyID can be get from `d`, RuleType is fixed, RuleID can be get from 'd', only the param of Cfg is required, if so, the total param number is 3